public/components/ContentPanel/ContentPanelActions.tsx

```diff
@@ -15,20 +15,46 @@
 
 import React from "react";
 import { EuiButton, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
+import { ModalConsumer } from "../Modal";
 
 interface ContentPanelActionsProps {
-  actions: { text: string; buttonProps?: object; flexItemProps?: object }[];
+  actions: {
+    text: string;
+    buttonProps?: object;
+    flexItemProps?: object;
+    modal?: {
+      onClickModal: (onShow: (component: any, props: object) => void) => () => void;
+    };
+  }[];
 }
 
 const ContentPanelActions: React.SFC<ContentPanelActionsProps> = ({ actions }) => (
   <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
-    {actions.map(({ text, buttonProps = {}, flexItemProps = {} }, index) => (
-      <EuiFlexItem {...flexItemProps} grow={false} key={index}>
+    {actions.map(({ text, buttonProps = {}, flexItemProps = {}, modal = null }, index) => {
```

Comment:
Note: I don't see a change where this component  is consumed and model is passed or not. Assume you have already have it.

Reply:
Yes, it will be part of a larger PR later, this was just adding the support of adding an action that opens a modal and verifying tests passed.